exercises/sum-of-multiples/example.rs

```diff
@@ -0,0 +1,17 @@
+use std::collections::BTreeSet;
+
+pub fn sum_of_multiples(limit: u32, factors: &Vec<u32>) -> u32 {
```

Comment:
by no means do I demand this, but I look at https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#slicing and see "In Rust, it's more common to pass slices as arguments rather than vectors when you just want to provide a read access." - so I suggest `&[u32]` here.

(Note that doing this should not require the test code to change in any way - the `vec!` in the test code can still stay)